code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,648 @@
+            required:
+              - componentName
+              - operatingSystem
+              - cpuArchitecture
+              - images
+              - networkInterfaces
+              - numOfInstances
+              - restartPolicy
+            properties:
+              componentName: 
+                type: string
+                pattern:  ^[A-Za-z0-9][A-Za-z0-9_]{6,62}[A-Za-z0-9]$
+                description: Must be a valid RFC 1035 label name.  Component name must be unique with an application
+              operatingSystem:
+                $ref: '#/components/schemas/OperatingSystem'
+              cpuArchitecture:
+                $ref: '#/components/schemas/CpuArchType'
+              images:
+                description: List of all images assosiated with the component. Images are uploaded or specified using UploadFile apis
+                type: array
+                items: 
+                  $ref: '#/components/schemas/FileId' 
+                minItems: 1
+              networkInterfaces:
+                description: Each application component exposes some ports either for external users or for inter component communication.
+                                    Application provider is required to specify which ports are to be exposed and the type of traffic that will flow through these ports.
+ 
+                type: array
+                items:                          
+                  type: object
+                  required:
+                    - interfaceId
+                    - protocol
+                    - port
+                    - network
+                    - interfaceName
+                    - visibilityType
+                  properties:
+                    interfaceId:
+                      type: string
+                      description:  Each Port and corresponding traffic protocol exposed by the component is identified by a name. 
+                                            Application client on user device requires this to uniquley idenify the interface. 
+                      pattern: ^[A-Za-z0-9][A-Za-z0-9_]{6,30}[A-Za-z0-9]$
+                    network:
+                      type: string
+                      pattern: ^[A-Za-z][A-Za-z0-9_]{6,30}[A-Za-z0-9]$
+                      description: Name of the network.  In case the application has to be assoisated with more then 1 network then app provider
+                                          must define the name of the network on which this interface has to be exposed.  This parameter is required only if 
+                                          the port has to be exposed on a specific network other then default.
+                    protocol:
+                      type: string
+                      enum:
+                        - TCP
+                        - UDP
+                        - HTTP/HTTPS
+                      description: Defines the IP transport communication protocol i.e., TCP, UDP or HTTP
```

Comment:
Note that:
- HTTP(s) aren't transport protocols.
- If we want to extend this to app protocols, I would suggest we leave room for less used protocols using an `ANY` option.

Reply:
I agree we should leave room for application level protocols

Reply:
Agree, `ANY` parameter was added